src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/CompiledPageActionDescriptorBuilder.cs

```diff
@@ -40,6 +40,7 @@ public static class CompiledPageActionDescriptorBuilder
                 FilterDescriptors = filters,
                 HandlerMethods = handlerMethods,
                 HandlerTypeInfo = applicationModel.HandlerType,
+                DeclaredModelTypeInfo = applicationModel.DeclaredModelType,
```

Comment:
Should we check here that `ModelType.IsAssignableFrom(DeclaredModelType)`? Otherwise this will blow up with a cryptic error at runtime if you do it wrong.

Reply:
Fair enough